Fixes for episodeDetails

mre opened the pull request:
Not all fields get sent by Apple; provide defaults as fallback.
This affects the code and the DB.

mre commented:
Corresponding DB deploy request: https://app.planetscale.com/services-openpodcast/openpodcast/deploy-requests/31

woolfg commented on db_schema/schema.sql:
```diff
@@ -156,11 +156,11 @@ DROP TABLE IF EXISTS appleEpisodeDetails;
-  aed_uniqueengagedlistenerscount INTEGER NOT NULL,
-  aed_uniquelistenerscount INTEGER NOT NULL,
-  aed_engagedplayscount INTEGER NOT NULL,
+  aed_playscount INTEGER,
```
why did you remove NOT NULL? do you set it to NULL, it should be just 0, right?

woolfg replied:
if we convince ajv to not only validate the schema but also edit the data, this should be obsolete (see other comment including the link to the option for ajv)

mre replied:
true; added the option and reverted the schema changes. Tested and works.

woolfg commented on src/schema/apple/episodeDetails.json:
```diff
@@ -38,17 +38,7 @@
           "type": "number",
           "default": 0
         }
-      },
```
I think we can add this again, if we convince ajv to not only validate the schema but also edit the data:
see https://ajv.js.org/options.html#usedefaults

maybe we can add this in `src/schema/SchemaValidator.ts`

mre replied:
true; added the option and reverted the schema changes. Tested and works

woolfg commented on fixtures/appleEpisodeDetailsSparse.json:
```diff
@@ -0,0 +1,62 @@
+    },
+    "episodePlayHistogram": [
+      {
+        "2865": 11.0
```
that looks really broken, 11 listeners until 2865? but yeah, I guess we have to deal with this.

woolfg reviewed (approved):
lgtm

mre commented:
Added the option to AJV and reverted the schema changes. They are no longer needed because AJV fills in the missing defaults. 👍 Tested and works with empty perf and non-empty perf data.

woolfg reviewed (approved):
looks still good to me, meeerge i
